src/array/array.jl

```diff
@@ -283,6 +289,12 @@ function DimArray(f::Function, dim::Dimension; name=Symbol(nameof(f), "(", name(
      DimArray(f.(val(dim)), (dim,); name)
 end
 
+checkdims(A::AbstractArray{<:Any,N}, dims::Tuple) where N =
```

Comment:
Should this have a unit test?

Reply:
Yeah probably, its just a test of `rebuild` currently.